### PR TITLE
Add compat data for ServiceWorkerMessageEvent

### PR DESCRIPTION
**Pull request by bunnybooboo**
https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerMessageEvent

as in #1002 notes for ESR might need editing.

### CODE REVIEW

File: api/ServiceWorkerMessageEvent.json [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+          },
+          "chrome": {
+            "version_added": "45"
+          },
+          "chrome_android": {
+            "version_added": "45"
+          },
+          "edge": {
+            "version_added": "17",
+            "notes": "Service workers is available in Microsoft Edge starting EdgeHTML 16 behind a flag."
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: "flags"

File: api/ServiceWorkerMessageEvent.json [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+          },
+          "edge_mobile": {
+            "version_added": null
+          },
+          "firefox": {
+            "version_added": "44",
+            "version_removed": "55",
+            "notes": "Service workers (and Push) have been disabled in the Firefox 45 and 52 Extended Support Releases (ESR)."
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: I would add a 2nd note saying that: "In Firefox 55 and later, the standard \<code>MessageEvent\</code> interface must be used instead." (only on the interface)

File: api/ServiceWorkerMessageEvent.json [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+            },
+            "chrome": {
+              "version_added": "45"
+            },
+            "chrome_android": {
+              "version_added": "45"
+            },
+            "edge": {
+              "version_added": "17",
+              "notes": "Service workers is available in Microsoft Edge starting EdgeHTML 16 behind a flag."
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: "flags"

File: api/ServiceWorkerMessageEvent.json [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: "flags"

File: api/ServiceWorkerMessageEvent.json [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: "flags"

File: api/ServiceWorkerMessageEvent.json [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: "flags"

File: api/ServiceWorkerMessageEvent.json [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: "flags"

File: api/ServiceWorkerMessageEvent.json [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+            },
+            "edge_mobile": {
+              "version_added": null
+            },
+            "firefox": {
+              "version_added": "44",
+              "notes": "Service workers (and Push) have been disabled in the Firefox 45 and 52 Extended Support Releases (ESR)."
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: "version_removed" is missing here.

File: api/ServiceWorkerMessageEvent.json [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+            },
+            "chrome": {
+              "version_added": "45"
+            },
+            "chrome_android": {
+              "version_added": "45"
+            },
+            "edge": {
+              "version_added": "17",
+              "notes": "Service workers is available in Microsoft Edge starting EdgeHTML 16 behind a flag."
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: "flags"

File: api/ServiceWorkerMessageEvent.json [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+            },
+            "edge_mobile": {
+              "version_added": null
+            },
+            "firefox": {
+              "version_added": "44",
+              "notes": "Service workers (and Push) have been disabled in the Firefox 45 and 52 Extended Support Releases (ESR)."
```
Comment (teoli2003) [inline api/ServiceWorkerMessageEvent.json]: "version_removed" is missing here.

File: api/ServiceWorkerMessageEvent.json (hunk 1 above)
Comment (bunnybooboo) [reply api/ServiceWorkerMessageEvent.json]: Hmm. As the data in the table & notes only refers to the above, how do I shape this as a flag? Previously (Fx example earlier today) I've had:

```
"type": "preference",  // known
"name": "dom.payments.request.enabled",  // unknown
"value_to_set": "true" // unknown
```
Does it get added as a note? Looking in Edge release notes, the flag is not clear to a newbie like me. :confused:

Comment (bunnybooboo) [reply api/ServiceWorkerMessageEvent.json]: ahh I think you answered this elsewhere. Wrapping my head around it. No need to respond to me now.

File: api/ServiceWorkerMessageEvent.json [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+            "version_removed": "55"
+          },
+          "ie": {
+            "version_added": false
+          },
+          "opera": {
+            "version_added": "24"
+          },
+          "opera_android": {
+            "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerMessageEvent.json]: This should be 24.

File: api/ServiceWorkerMessageEvent.json [hunk 12]
```diff
@@ -0,0 +1,367 @@
...
+              "version_removed": "55"
+            },
+            "ie": {
+              "version_added": false
+            },
+            "opera": {
+              "version_added": "24"
+            },
+            "opera_android": {
+              "version_added": null
```
Comment (jpmedley) [inline api/ServiceWorkerMessageEvent.json]: 24

File: api/ServiceWorkerMessageEvent.json [hunk 13]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (jpmedley) [inline api/ServiceWorkerMessageEvent.json]: 24

File: api/ServiceWorkerMessageEvent.json [hunk 14]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (jpmedley) [inline api/ServiceWorkerMessageEvent.json]: 24

File: api/ServiceWorkerMessageEvent.json [hunk 15]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (jpmedley) [inline api/ServiceWorkerMessageEvent.json]: 24

File: api/ServiceWorkerMessageEvent.json [hunk 16]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (jpmedley) [inline api/ServiceWorkerMessageEvent.json]: 24

File: api/ServiceWorkerMessageEvent.json [hunk 17]
```diff
@@ -0,0 +1,367 @@
...
```
Comment (jpmedley) [inline api/ServiceWorkerMessageEvent.json]: 24

File: api/ServiceWorkerMessageEvent.json (hunk 1 above)
Comment (teoli2003) [reply api/ServiceWorkerMessageEvent.json]: Yes, this is it (indentation not correct):

"edge": [ {
"version_added": "17"
},
{
"version_added: "16",
"flags": [ { 
   "type": "preference",
    "name": "Enable service workers"
 } ]
}
]

File: api/ServiceWorkerMessageEvent.json (hunk 2 above)
Comment (bunnybooboo) [reply api/ServiceWorkerMessageEvent.json]: Actioning the rest of this now

**Comment by Elchi3** [comment]
Looks good to me. If I see it correctly, Joe's comments have not been addressed yet, but otherwise this is good to go.

**Pull request merged**